File: ios/Plugin/AutoStartManagerPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(AutoStartManagerPlugin)
public class AutoStartManagerPlugin: CAPPlugin {
    private let implementation = AutoStartManager()
    private let dateTimeCheck = DateTimeCheck()  // Instance dari DateTimeCheck

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }

    // Fungsi untuk mengecek apakah pengaturan waktu otomatis aktif
    @objc func checkAutomaticDateTime(_ call: CAPPluginCall) {
        dateTimeCheck.checkIfAutomaticDateTimeIsEnabled { isAutomatic in
            call.resolve([
                "isAutomatic": isAutomatic
            ])
        }
    }
}

File: android/src/main/java/io/github/asephermann/plugins/autostartmanager/AutoStartManagerPlugin.kt
package io.github.asephermann.plugins.autostartmanager

import com.getcapacitor.JSObject
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin

@CapacitorPlugin(name = "AutoStartManager")
class AutoStartManagerPlugin : Plugin() {
    private val implementation = AutoStartManager()
    @PluginMethod
    fun echo(call: PluginCall) {
        val value = call.getString("value")
        val ret = JSObject()
        ret.put("value", value?.let { implementation.echo(it) })
        call.resolve(ret)
    }
}